.github/workflows/r_test_and_check.yml

```diff
@@ -0,0 +1,120 @@
+name: R Check and Unit Tests
```

Comment:
is it possible to separate test and r check in separate pipelines?

Reply:
It is possible, but I think unit tests only is not enough check. 
1. R is an interpreted language, you can pet syntax error for sometime without getting detected. R Check will detect this among many other issues.
2. If we separate, I'm afraid R check to be undervalued by developers (they just say, "Hey, the tests are all passed! Who cares about R check?" Eventhough you have breaking changes (example breaks and syntax error)

Test + check takes about 6 minutes.
Since the latest check has already shown unit tests detail, I can even suggest to remove the normal tests from the pipeline to save about 1 minutes.

Reply:
what about the build package test? that we had a yml pipeline for it?

Reply:
We did have it. As I explained in other comment, r_test_and_check.yml also covers that scenario since rcmdcheck does build the package first, and do check against the build package: 
[https://github.com/ForestForesight/ForestForesight/actions/runs/12085403473/job/33702559410](── R CMD build ─────────────────
* checking for file ‘.../DESCRIPTION’ ... OK
* preparing ‘ForestForesight’:
* checking DESCRIPTION meta-information ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
Removed empty directory ‘ForestForesight/tests/testthat/_snaps’
  NB: this package now depends on R (>= 3.5.0)
  WARNING: Added dependency on R >= 3.5.0 because serialized objects in
  serialize/load version 3 cannot be read in older versions of R.
  File(s) containing such objects:
    ‘ForestForesight/data/countries.rda’
    ‘ForestForesight/data/degree_polygons.rda’
    ‘ForestForesight/data/feature_metadata.rda’
    ‘ForestForesight/data/gfw_tiles.rda’
    ‘ForestForesight/tests/test_data/models/South-Eastern Asia
    1/South-Eastern Asia 1.rda’
* building ‘ForestForesight_4.0.0.tar.gz’
Warning in utils::tar(filepath, pkgname, compression = compression, compression_level = 9L,  :
  storing paths of more than 100 bytes is not portable:
  ‘ForestForesight/tests/test_data/preprocessed/groundtruth/10N_110E/10N_110E_2023-01-01_groundtruth6m.tif’
Warning in utils::tar(filepath, pkgname, compression = compression, compression_level = 9L,  :
  storing paths of more than 100 bytes is not portable:
  ‘ForestForesight/tests/test_data/preprocessed/groundtruth/10N_110E/10N_110E_2023-02-01_groundtruth6m.tif’
Warning in utils::tar(filepath, pkgname, compression = compression, compression_level = 9L,  :
  storing paths of more than 100 bytes is not portable:
  ‘ForestForesight/tests/test_data/preprocessed/input/10N_110E/10N_110E_2021-01-01_initialforestcover.tif’
── R CMD check ───────────────────
* using log directory ‘/tmp/RtmpubRoGp/file1eaa[55](https://github.com/ForestForesight/ForestForesight/actions/runs/12085403473/job/33702559410#step:10:56)633577/ForestForesight.Rcheck’
* using R version 4.4.2 (2024-10-31)
* using platform: x86_64-pc-linux-gnu
* R was compiled by
    gcc (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
    GNU Fortran (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
* running under: Ubuntu 22.04.5 LTS
* using session charset: UTF-8
* using option ‘--as-cran’
* checking for file ‘ForestForesight/DESCRIPTION’ ... OK
* checking extension type ... Package
* this is package ‘ForestForesight’ version ‘4.0.0’
* package encoding: UTF-8
* checking CRAN incoming feasibility ... NOTE
Maintainer: ‘Jonas van Duijvenbode <jduijvenbode@wwf.nl>’
New submission
Non-FOSS package license (GPL-3 + COPYING file)
The Title field should be in title case. Current version is:
‘Package created to preprocess, model, predict and analyze deforestation according to the WWF-NL ForestForesight principles.’
In title case that is:
‘Package Created to Preprocess, Model, Predict and Analyze Deforestation According to the WWF-NL ForestForesight Principles.’
The Description field should not start with the package name,
  'This package' or similar.
The Date field is over a month old.
Size of tarball: 13203775 bytes
* checking package namespace information ... OK
* checking package dependencies ... NOTE
Depends: includes the non-default packages:
  'terra', 'sf', 'xgboost', 'httr', 'lubridate', 'smoothr', 'crayon',
  'aws.s3'
Adding so many packages to the search path is excessive and importing
selectively is preferable.
* checking if this is a source package ... OK
* checking if there is a namespace ... OK
* checking for executable files ... OK
* checking for hidden files and directories ... NOTE)